src/vendor/OktaAuth.js

```diff
@@ -365,98 +382,79 @@ function deleteCookie(name) {
-  return addStateToken(trans);
+function getStateToken(res) {
+  return addStateToken(res);
 }
 
 function AuthTransaction(sdk, res) {
   if (res) {
-    this.response = res;
-
-    if (res.status) {
-      this.status = res.status;
-      var methods = STATE_MAP[res.status](sdk, this);
-      extend(this, methods);
+    this.data = res;
+    extend(this, flattenEmbedded(sdk, res, res, {}));
+    delete this.stateToken;
+
+    // RECOVERY_CHALLENGE has some responses without _links.
+    // Without _links, we emulate cancel to make it intuitive
+    // to return to the starting state. We may remove this
+    // when OKTA-75434 is resolved
+    if (res.status === 'RECOVERY_CHALLENGE' && !res._links) {
+      this.cancel = function() {
+        return new Q(new AuthTransaction(sdk));
+      };
     }
   }
 }
 
 function postToTransaction(sdk, uri, options) {
-  return sdk.post(uri, options)
+  return post(sdk, uri, options)
     .then(function(res) {
       return new AuthTransaction(sdk, res);
     });
 }
 
-function getCancelFn(sdk, trans, ref) {
-  function cancel() {
-    if (ref) {
-      ref.isPolling = false;
-    }
-    var cancelLink = getLink(trans.response, 'cancel');
-    return postToTransaction(sdk, cancelLink.href, getStateToken(trans));
-  }
-  return cancel;
-}
-
-function getPreviousFn(sdk, trans, ref) {
-  function previous() {
-    if (ref) {
-      ref.isPolling = false;
-    }
-    var previousLink = getLink(trans.response, 'prev');
-    return postToTransaction(sdk, previousLink.href, getStateToken(trans));
-  }
-  return previous;
-}
-
-function getPollFn(sdk, trans, ref) {
+function getPollFn(sdk, res, ref) {
   return function (delay) {
     if (!delay && delay !== 0) {
       delay = DEFAULT_POLLING_DELAY;
     }
 
     // Get the poll function
-    var pollLink = getLink(trans.response, 'next', 'poll');
+    var pollLink = getLink(res, 'next', 'poll');
     function pollFn() {
-      return sdk.post(pollLink.href, getStateToken(trans), true, true);
+      return post(sdk, pollLink.href, getStateToken(res), true, true);
     }
 
-    // Exposed to stop any polling in tests
-    pollFn.__ref = ref;
-
     ref.isPolling = true;
 
     var retryCount = 0;
     var recursivePoll = function () {
 
       // If the poll was manually stopped during the delay
       if (!ref.isPolling) {
-        return Q.resolve(trans);
+        return Q.reject(new AuthPollStopError());
       }
 
       return pollFn()
-        .then(function (res) {
+        .then(function (pollRes) {
           // Reset our retry counter on success
           retryCount = 0;
 
           // If we're still waiting
-          if (res.factorResult && res.factorResult === 'WAITING') {
+          if (pollRes.factorResult && pollRes.factorResult === 'WAITING') {
 
             // If the poll was manually stopped while the pollFn was called
             if (!ref.isPolling) {
-              return trans;
+              throw new AuthPollStopError();
```

Comment:
We now through an `AuthPollStopError` when we stop polling instead of returning the transaction